cmd/misc/main.go

```diff
@@ -1233,3 +1240,294 @@ OUTER:
+		}
+	}
+
+	if len(onlySyncValidatorData) == 0 {
+		return nil // no sync committee yet skip
+	}
+
+	logrus.Infof("statistics data collection for day %v completed", day)
+
+	// calculate cl income data & update totals
+	for _, data := range onlySyncValidatorData {
+
+		previousDayData := &types.ValidatorStatsTableDbRow{
+			ValidatorIndex: math.MaxUint64,
+		}
+
+		if day == 0 {
+			previousDayData.ValidatorIndex = data.ValidatorIndex
+		}
+
+		if data.ValidatorIndex < uint64(len(statisticsData1d)) && day > 0 {
+			previousDayData = statisticsData1d[data.ValidatorIndex]
+		}
+
+		if data.ValidatorIndex != previousDayData.ValidatorIndex {
+			return fmt.Errorf("logic error when retrieving previous day data for validator %v (%v wanted, %v retrieved)", data.ValidatorIndex, data.ValidatorIndex, previousDayData.ValidatorIndex)
+		}
+
+		// update sync total
+		data.ParticipatedSyncTotal = previousDayData.ParticipatedSyncTotal + data.ParticipatedSync
+		data.MissedSyncTotal = previousDayData.MissedSyncTotal + data.MissedSync
+		data.OrphanedSyncTotal = previousDayData.OrphanedSyncTotal + data.OrphanedSync
+	}
+
+	var statisticsDataToday []*types.ValidatorStatsTableDbRow
+	if dryRun {
+		var err error
+		statisticsDataToday, err = db.GatherStatisticsForDay(int64(day)) // convert to int64 to avoid underflows
+		if err != nil {
+			return fmt.Errorf("error in GatherPreviousDayStatisticsData: %w", err)
+		}
+	}
+
+	tx, err := db.WriterDb.Beginx()
+	if err != nil {
+		return fmt.Errorf("error retrieving raw sql connection: %w", err)
+	}
+	defer tx.Rollback()
+
+	logrus.Infof("updating statistics data into the validator_stats table %v | %v", len(onlySyncValidatorData), len(validatorData))
+
+	for _, data := range onlySyncValidatorData {
+		if dryRun {
+			logrus.Infof(
+				"validator %v: participated sync: %v -> %v, missed sync: %v -> %v, orphaned sync: %v -> %v, total participated: %v -> %v, total missed sync: %v -> %v, total orphaned sync: %v -> %v",
+				data.ValidatorIndex, statisticsDataToday[data.ValidatorIndex].ParticipatedSync, data.ParticipatedSync, statisticsDataToday[data.ValidatorIndex].MissedSync, data.MissedSync, statisticsDataToday[data.ValidatorIndex].OrphanedSync,
+				data.OrphanedSync, statisticsDataToday[data.ValidatorIndex].ParticipatedSyncTotal, data.ParticipatedSyncTotal, statisticsDataToday[data.ValidatorIndex].MissedSyncTotal, data.MissedSyncTotal, statisticsDataToday[data.ValidatorIndex].OrphanedSyncTotal, data.OrphanedSyncTotal,
+			)
+		} else {
+			tx.Exec(`
+				UPDATE validator_stats set
+				participated_sync = $1,
+				participated_sync_total = $2,
+				missed_sync = $3,
+				missed_sync_total = $4,
+				orphaned_sync = $5,
+				orphaned_sync_total = $6
+				WHERE day = $7 AND validatorindex = $8`,
+				data.ParticipatedSync, data.ParticipatedSyncTotal,
+				data.MissedSync, data.MissedSyncTotal,
+				data.OrphanedSync, data.OrphanedSyncTotal,
+				data.Day, data.ValidatorIndex)
+		}
```

Comment:
Is doing 1 update query per validator efficient? Why not do something with `INSERT VALUES`?

Reply:
It only updates validators with sync committee per day, so that's 512 updates per day. Which is fine performance wise and the code runs once so I did not see the need to optimize it any further. We can keep it simple and more readable therefore.